File: test/fixtures/dart/j_patch_file_with_data.dart
import 'package:http/http.dart' as http;

void main() async {
  final url = Uri.parse('http://localhost:28139/patch');

  final req = http.MultipartRequest('PATCH', url)
    ..files.add(await http.MultipartFile.fromPath(
      'file1', './test/fixtures/curl_commands/delete.sh'))
    ..fields['form1'] = 'form+data+1'
    ..fields['form2'] = 'form_data_2';

  final stream = await req.send();
  final res = await http.Response.fromStream(stream);
  final status = res.statusCode;
  if (status != 200) throw Exception('http.send error: statusCode= $status');

  print(res.body);
}

File: test/fixtures/r/j_patch_file_with_data.r
library(httr)

files = list(
  file1 = upload_file("./test/fixtures/curl_commands/delete.sh"),
  form1 = "form+data+1",
  form2 = "form_data_2"
)

res <- httr::PATCH(url = "http://localhost:28139/patch", body = files, encode = "multipart")
